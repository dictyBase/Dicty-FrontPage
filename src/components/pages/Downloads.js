// @flow
import React from "react"
import { Flex, Box } from "rebass"
import FontAwesome from "react-fontawesome"
import { Banner, Header, Hdrtxt } from "styles/EditablePageStyles"

/**
 * This displays the Dicty downloads page.
 */

const Downloads = () => (
  <div>
    <Banner style={{ backgroundColor: "#BCF5A9" }}>
      <Header style={{ backgroundColor: "#BCF5A9" }}>Downloads</Header>
      <Hdrtxt>
        The central collection of downloadable material from the dictyBase
      </Hdrtxt>
      <FontAwesome name="arrow-circle-down" />
      <br />
      <br />
      <em>Coming soon</em>
    </Banner>
    <br />
    <Flex justify="center">
      <Box>
        <h3 style={{ textAlign: "center" }}>dictyBase</h3>
        <p style={{ textAlign: "center" }}>
          <strong>
            {" "}
            Center for Genetic Medicine
            <br /> Feinberg School of Medicine
            <br /> Northwestern University
            <br />{" "}
          </strong>
        </p>
        <p> The Arthur Rubloff Building 420 East Superior Street </p>
      </Box>
    </Flex>
  </div>
)

export default Downloads
