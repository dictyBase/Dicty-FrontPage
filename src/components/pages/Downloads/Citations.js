// @flow
import React from "react"
import { withStyles } from "@material-ui/core/styles"

const styles = theme => ({
  root: {
    fontSize: "0.9em",
  },
  topLine: {
    color: "#a7221e",
  },
  link: {
    textDecoration: "none",
  },
})

type Props = {
  /** Material-UI styling */
  classes: Object,
  /** Citations data */
  citation: {
    /** Authors of paper */
    authors: String,
    /** Title of paper */
    title: String,
    /** Journal for this citation */
    journal: String,
    /** Link to PubMed */
    link: String,
  },
}

/**
 * Displays the citation above the downloads table.
 */

const Citations = (props: Props) => {
  const { classes, citation } = props

  return (
    <div className={classes.root}>
      <h3 className={classes.topLine}>Please cite:</h3>
      <p>
        <strong>{citation.authors}</strong>
        {"  "}
        {citation.title}
        {"  "}
        <em>{citation.journal}</em>
        {"  "}
        <a
          href={citation.link}
          target="_blank"
          rel="noopener noreferrer"
          className={classes.link}>
          [Pubmed]
        </a>
      </p>
    </div>
  )
}

export default withStyles(styles)(Citations)
