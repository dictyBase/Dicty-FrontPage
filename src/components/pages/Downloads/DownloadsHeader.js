import React from "react"

const DownloadsHeader = () => (
  <div>
    <h1>
      <center>dictyBase Downloads</center>
    </h1>
    <h3>
      <center>
        The central collection of downloadable material from the dictyBase
      </center>
    </h3>
  </div>
)

export default DownloadsHeader
