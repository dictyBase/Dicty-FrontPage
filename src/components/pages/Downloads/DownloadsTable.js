import React, { Fragment } from "react"
import { withStyles } from "@material-ui/core/styles"
import Table from "@material-ui/core/Table"
import TableBody from "@material-ui/core/TableBody"
import TableCell from "@material-ui/core/TableCell"
import TableHead from "@material-ui/core/TableHead"
import TableRow from "@material-ui/core/TableRow"
import Paper from "@material-ui/core/Paper"

const styles = theme => ({
  root: {
    width: "100%",
    marginTop: theme.spacing.unit * 3,
    overflowX: "auto",
  },
  table: {
    minWidth: 700,
  },
})

const DownloadsTable = props => {
  const { classes, data } = props

  return (
    <Paper className={classes.root}>
      <Table className={classes.table}>
        {data.data.map(section => (
          <Fragment key={section.attributes.title}>
            <TableHead>
              <TableRow>
                <TableCell>{section.attributes.title}</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {section.attributes.items.map(row => (
                <TableRow key={row.title}>
                  <TableCell component="th" scope="row">
                    {row.title}
                  </TableCell>
                  <TableCell>
                    <a href={row.url}>Download</a>
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Fragment>
        ))}
      </Table>
    </Paper>
  )
}

export default withStyles(styles)(DownloadsTable)
