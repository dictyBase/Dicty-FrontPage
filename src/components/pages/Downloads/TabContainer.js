import React from "react"
import Typography from "@material-ui/core/Typography"

type Props = {
  children: any,
}

const TabContainer = (props: Props) => (
  <Typography component="div">{props.children}</Typography>
)

export default TabContainer
