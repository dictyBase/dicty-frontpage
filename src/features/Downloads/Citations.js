import React from "react"
import { withStyles } from "@material-ui/core/styles"

const styles = (theme) => ({
  root: {
    fontSize: "0.9em",
  },
  topLine: {
    color: "#a7221e",
  },
  link: {
    textDecoration: "none",
  },
})

/**
 * Displays the citation above the downloads table.
 */

const Citations = ({ classes, citations }) => (
  <div className={classes.root}>
    <h3 className={classes.topLine}>Please cite:</h3>
    {citations.map((citation) => (
      <p key={citation.title}>
        <strong>{citation.authors}</strong>
        {"  "}
        {citation.title}
        {"  "}
        <em>{citation.journal}</em>
        {"  "}
        <a
          href={citation.link}
          target="_blank"
          rel="noopener noreferrer"
          className={classes.link}>
          [Pubmed]
        </a>
      </p>
    ))}
  </div>
)

export default withStyles(styles)(Citations)
