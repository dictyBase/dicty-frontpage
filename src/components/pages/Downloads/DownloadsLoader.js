import React from "react"
import Skeleton, { SkeletonTheme } from "react-loading-skeleton"
import Grid from "@material-ui/core/Grid"
import DownloadsHeader from "./DownloadsHeader"

/**
 * Skeleton loading component displayed while downloads data is being fetched.
 */

const DownloadsLoader = () => (
  <Grid container justify="center">
    <Grid item xs={8}>
      <DownloadsHeader />
      <SkeletonTheme color="#d1d1d1">
        <br />
        <Skeleton count={10} />
      </SkeletonTheme>
    </Grid>
  </Grid>
)

export default DownloadsLoader
