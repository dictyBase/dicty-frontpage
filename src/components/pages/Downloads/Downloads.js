// @flow
import React, { Component } from "react"
import { Helmet } from "react-helmet"
import { connect } from "react-redux"
import Grid from "@material-ui/core/Grid"
import AppBar from "@material-ui/core/AppBar"
import Tabs from "@material-ui/core/Tabs"
import Tab from "@material-ui/core/Tab"
import Typography from "@material-ui/core/Typography"
import { MuiThemeProvider } from "@material-ui/core/styles"
import Citations from "./Citations"
import DownloadsTable from "./DownloadsTable"
import DownloadsHeader from "./DownloadsHeader"
import DownloadsLoader from "./DownloadsLoader"
import withDataFetching from "components/common/withDataFetching"
import MuiTheme from "components/common/MuiTheme"
import { fetchDownloadTabs, changeTab } from "actions/downloads"

type Props = {
  /** The downloads slice of the state */
  downloads: Object,
  /** Action creator to fetch the download tabs data */
  fetchDownloadTabs: Function,
  /** Action that changes the value of the selected tab */
  changeTab: Function,
}

/**
 * This displays the Dicty downloads page.
 */

class Downloads extends Component<Props> {
  handleChange = (event: SyntheticEvent<>, value: string) => {
    const { changeTab } = this.props

    changeTab(value)
  }

  generateTabs = (json: Object) => {
    const tabs = json.data.map(item => (
      <Tab
        value={item.id}
        label={item.attributes.scientific_name}
        key={item.id}
      />
    ))
    return tabs
  }

  generateTabContainers = (json: Object) => {
    const { downloads } = this.props

    const tabContent = json.data.map(item => {
      if (item.id === json.currentTab) {
        return (
          <Typography component="div" key={item.id}>
            <Citations citations={item.attributes.citations} />
            {downloads[item.id] && (
              <DownloadsTable data={downloads[item.id].data} />
            )}
          </Typography>
        )
      }
      return null
    })
    return tabContent
  }

  render() {
    const { downloads } = this.props

    // return null until downloads data is not empty
    if (Object.entries(downloads).length === 0) {
      return null
    }

    return (
      <MuiThemeProvider theme={MuiTheme}>
        <Helmet>
          <title>dictyBase Downloads</title>
          <meta
            name="description"
            content="dictyBase Downloads - the central collection of downloadable material from dictyBase"
          />
        </Helmet>
        <Grid container justify="center">
          <Grid item xs={8}>
            <DownloadsHeader />
            <AppBar position="static">
              <Tabs
                value={downloads.currentTab}
                onChange={this.handleChange}
                variant="scrollable"
                scrollButtons="auto">
                {this.generateTabs(downloads)}
              </Tabs>
            </AppBar>
            {this.generateTabContainers(downloads)}
          </Grid>
        </Grid>
      </MuiThemeProvider>
    )
  }
}

const mapStateToProps = ({ downloads }) => ({ downloads })

const DownloadsWithFetch = withDataFetching(
  fetchDownloadTabs,
  "downloads",
  DownloadsLoader,
)(Downloads)

export { Downloads }
export default connect(mapStateToProps, { changeTab })(DownloadsWithFetch)
