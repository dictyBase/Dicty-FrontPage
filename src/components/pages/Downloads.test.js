import React from "react"
import toJson from "enzyme-to-json"
import { shallow } from "enzyme"
import "jest-styled-components"
import Downloads from "./Downloads"

test("matching a snapshot of Downloads", () => {
  const subject = shallow(<Downloads />)
  expect(toJson(subject)).toMatchSnapshot()
})
