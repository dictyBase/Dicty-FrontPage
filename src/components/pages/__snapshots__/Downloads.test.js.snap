// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Downloads 1`] = `
<div>
  <styled.div
    style={
      Object {
        "backgroundColor": "#BCF5A9",
      }
    }
  >
    <styled.h1
      style={
        Object {
          "backgroundColor": "#BCF5A9",
        }
      }
    >
      Downloads
    </styled.h1>
    <styled.p>
      The central collection of downloadable material from the dictyBase
    </styled.p>
    <FontAwesome
      name="arrow-circle-down"
    />
    <br />
    <br />
    <em>
      Coming soon
    </em>
  </styled.div>
  <br />
  <Flex
    justify="center"
  >
    <Box>
      <h3
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        dictyBase
      </h3>
      <p
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <strong>
           
          Center for Genetic Medicine
          <br />
           Feinberg School of Medicine
          <br />
           Northwestern University
          <br />
           
        </strong>
      </p>
      <p>
         The Arthur Rubloff Building 420 East Superior Street 
      </p>
    </Box>
  </Flex>
</div>
`;
